Sources/SwiftGenKit/Utils/Metadata.swift

```diff
@@ -36,32 +36,35 @@ enum Metadata {
   /// - Parameter data: The value to describe
   /// - Returns: Dictionary with type information about the value (for Stencil context)
   static func generate(for data: Any) -> [String: Any] {
-    switch data {
-    case is String:
+    let dataType = type(of: data)
+
+    if data is String {
```

Comment:
I think it could be worth a code comment to explain (iiuc) that this code uses `data is X` when we want toll-free-bridging to apply but `dataType == X.self` when we want to avoid TFB like for Int/Bool/… which could be misinterpreted and mis-bridged when coming in as NSNumber.